numpydoc/xref.py

```diff
@@ -81,6 +81,7 @@
     "np.ndarray": "numpy.ndarray",
     "array-like": ":term:`numpy:array_like`",
     "array_like": ":term:`numpy:array_like`",
+    "ArrayLike": ":term:`numpy.typing.ArrayLike`",
```

Comment:
```suggestion
    "ArrayLike": ":term:`numpy:array_like`",
```

Here's the description of the term [`array_like`](https://numpy.org/devdocs/glossary.html#term-array_like) vs. the description of the term [`typing.ArrayLike`](https://numpy.org/devdocs/reference/typing.html#arraylike).

I still think the former is likely what you'd want to link to for users: it describes what `array_like` means, as opposed to how the annotation is expected to behave in a static typing context.

Not a blocker though, feel free to do as you see fit!